esp/services/ws_dfu/ws_dfuService.hpp

```diff
@@ -30,6 +30,42 @@
 #include "fvrelate.hpp"
 #include "dadfs.hpp"
 
+#include <list>
+
+#define PAGE_CACHE_TIMEOUT (1000*60*10)
+
+struct IterateFilesCacheElement: public CInterface, implements IInterface
+{
+    IMPLEMENT_IINTERFACE;
+    IterateFilesCacheElement(__int64 _cacheHint, unsigned _cacheFrom, unsigned _cacheTo, unsigned _totalFiles)
+        : cacheHint(_cacheHint), cacheFrom(_cacheFrom), cacheTo(_cacheTo), totalFiles(_totalFiles)
+    {
+        due = msTick()+PAGE_CACHE_TIMEOUT;
+    }
+
+    __int64 cacheHint;
+    unsigned cacheFrom;
+    unsigned cacheTo;
+    unsigned totalFiles;
+
+    unsigned due;
+};
+
+struct IterateFilesCache: public CInterface, implements IInterface
+{
+    IMPLEMENT_IINTERFACE;
+
+    IterateFilesCache(size32_t _cacheSize=0): cacheSize(_cacheSize){}
+    IterateFilesCacheElement* lookup(IEspContext &context, __int64 _cacheHint);
+
+    void add(__int64 _cacheHint, unsigned _cacheFrom, unsigned _cacheTo, unsigned _totalFiles);
+    void remove(__int64 _cacheHint);
+
+    std::list<Linked<IterateFilesCacheElement> > cache;
```

Comment:
Unusual to mix std library with jlib constructs like this.
I think it works, but internally at least we should be using jlib constructs consistently. e.g. a CIArrayOf<> or similar here.